types/net/net.d.ts

```diff
@@ -0,0 +1,70 @@
+declare module 'stripe' {
+  namespace Stripe {
+    export namespace HttpClient {
+      /**
+       * Abstract representation of an HTTP response. This is an experimental
+       * interface and is not yet stable.
+       */
+      interface Response {
+        /** The numeric HTTP status code for the response. */
+        getStatusCode(): number;
+
+        /** The response headers. */
+        getHeaders(): {[key: string]: string};
+
+        /** This returns the underlying raw response object for the client. */
+        getRawResponse(): any; //eslint-disable-line @typescript-eslint/no-explicit-any
+
+        /**
+         * This returns the content as a stream. This is implementation-specific
+         * and so we cannot use a given type as different platforms have
+         * different stream primitives.The expectation is that content
+         * will not have been buffered into memory at this point by the client.
+         *
+         * The streamCompleteCallback should be invoked by the response
+         * implementation when the stream has been consumed.
+         */
+        toStream(streamCompleteCallback: () => void): any; //eslint-disable-line @typescript-eslint/no-explicit-any
+
+        /**
+         * Converts the response content into a JSON object, failing if JSON
+         * couldn't be parsed.
+         */
+        toJSON(): Promise<object>;
+      }
+    }
+
+    /**
+     * Encapsulates the logic for issuing a request to the Stripe API. This is
+     * an experimental interface and is not yet stable.
+     */
+
+    export interface HttpClient {
+      makeRequest(
+        host: string,
+        port: string | number,
+        path: string,
+        method: 'GET' | 'POST' | 'PUT' | 'DELETE',
+        headers: object,
+        requestData: string | null,
+        protocol: Stripe.HttpProtocol,
+        timeout: number
+      ): Promise<HttpClient.Response>;
+    }
+
+    export class NodeHttpClient implements HttpClient {
+      constructor(agent?: Stripe.HttpAgent);
+
+      makeRequest(
+        host: string,
+        port: string | number,
+        path: string,
+        method: 'GET' | 'POST' | 'PUT' | 'DELETE',
+        headers: {[key: string]: string},
+        requestData: string | null,
+        protocol: Stripe.HttpProtocol,
+        timeout: number
+      ): Promise<HttpClient.Response>;
```

Comment:
I'm not a Typescript expert, but is this declaration required when it's explicitly saying that it implements the `HttpClient` interface?

Also is it important that the headers are a `{[key: string]: string}` but an `object` in the interface?

Reply:
> I'm not a Typescript expert, but is this declaration required when it's explicitly saying that it implements the HttpClient interface?

Unfortunately yes. See [this playground](https://www.typescriptlang.org/play?ssl=5&ssc=1&pln=5&pc=24#code/JYOwLgpgTgZghgYwgAgILIN4ChnICYD2AygQLYRgAWoA5gBQAeAXMiAK6kBG0AlMiwDcCwPAG4sAXyxYEAGzgBnBcgBCyYKQAOsiOXDL0GCUA) as an example. If we remove it, we get the error `Class 'NodeHttpClient' incorrectly implements interface 'HttpClient'.
  Property 'makeRequest' is missing in type 'NodeHttpClient' but required in type 'HttpClient'.

> Also is it important that the headers are a {[key: string]: string} but an object in the interface?

Good catch, that was a stale type. Both should be `object`.
`